File: experiments/adrianco/experiment-56-terra-alllang/brazil/runs/agent=codex_effort=default_language=typescript_model=gpt-5.6-terra_prompt=neutral/rep1/src/csv.ts
/** Minimal RFC-4180 compatible parser for the supplied Kaggle CSV files. */
export function parseCsv(text: string): Record<string, string>[] {
  const rows: string[][] = [];
  let row: string[] = [], field = "", quoted = false;
  for (let i = 0; i < text.length; i++) {
    const char = text[i];
    if (quoted) {
      if (char === '"' && text[i + 1] === '"') { field += char; i++; }
      else if (char === '"') quoted = false;
      else field += char;
    } else if (char === '"') quoted = true;
    else if (char === ',') { row.push(field); field = ""; }
    else if (char === '\n') { row.push(field.replace(/\r$/, "")); rows.push(row); row = []; field = ""; }
    else field += char;
  }
  if (field || row.length) { row.push(field.replace(/\r$/, "")); rows.push(row); }
  const headers = (rows.shift() ?? []).map((h) => h.replace(/^\uFEFF/, "").trim());
  return rows.filter((r) => r.some(Boolean)).map((r) => Object.fromEntries(headers.map((h, i) => [h, (r[i] ?? "").trim()])));
}

File: experiments/adrianco/experiment-56-terra-alllang/brazil/runs/agent=codex_effort=default_language=java_model=gpt-5.6-terra_prompt=neutral/rep1/src/main/java/com/brazilsoccer/mcp/Csv.java
package com.brazilsoccer.mcp;

import java.io.*;import java.nio.charset.StandardCharsets;import java.nio.file.*;import java.util.*;

/** Small RFC-4180-style reader; datasets contain quoted commas and UTF-8 text. */
final class Csv {
  static List<Map<String,String>> read(Path file) throws IOException {
    try (Reader r = Files.newBufferedReader(file, StandardCharsets.UTF_8)) {
      List<List<String>> rows = parse(r); if (rows.isEmpty()) return List.of();
      List<String> header = rows.remove(0); if (!header.isEmpty()) header.set(0, header.get(0).replace("\uFEFF", ""));
      List<Map<String,String>> result = new ArrayList<>();
      for (List<String> row: rows) { if (row.stream().allMatch(String::isEmpty)) continue; Map<String,String> m=new LinkedHashMap<>(); for(int i=0;i<header.size();i++)m.put(header.get(i),i<row.size()?row.get(i):""); result.add(m); }
      return result;
    }
  }
  private static List<List<String>> parse(Reader input) throws IOException {
    List<List<String>> rows=new ArrayList<>(); List<String> row=new ArrayList<>(); StringBuilder cell=new StringBuilder(); boolean quote=false; int c;
    while((c=input.read())!=-1){ char ch=(char)c; if(quote){if(ch=='\"'){input.mark(1);int next=input.read();if(next=='\"')cell.append('\"');else{quote=false;if(next!=-1)input.reset();}}else cell.append(ch);}else if(ch=='\"')quote=true;else if(ch==','){row.add(cell.toString().trim());cell.setLength(0);}else if(ch=='\n'){row.add(cell.toString().trim());cell.setLength(0);rows.add(row);row=new ArrayList<>();}else if(ch!='\r')cell.append(ch); }
    if(quote) throw new IOException("Unclosed quote in CSV"); if(!row.isEmpty()||cell.length()>0){row.add(cell.toString().trim());rows.add(row);} return rows;
  }
}
